test/boost/sstable_compaction_test.cc

```diff
@@ -184,17 +184,45 @@ static std::unique_ptr<strategy_control> make_strategy_control_for_test(bool has
     return std::make_unique<strategy_control_for_test>(has_ongoing_compaction);
 }
 
+class compaction_manager_for_testing {
+    lw_shared_ptr<compaction_manager> _cm;
+public:
+    explicit compaction_manager_for_testing(bool enabled = true) : _cm(make_lw_shared<compaction_manager>(compaction_manager::for_testing_tag{}))
+    {
```

Comment:
do you think we can reuse compaction_manager_test in lib/sstable_utils.hh?

Reply:
I see no reason why not, will change.

Reply:
done in v2